parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -85,6 +89,7 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
     return footerMetrics(metadata, fieldMetrics, metricsConfig, null);
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Decided to not refactor this method (at least not in this PR) since in this method we are essentially updating all column stats, and by moving logic directly out of the method we need to pass in those 6 stats map with some extra info, which could easily make the helper method exceeding 10 arguments.